src/main/java/ch/wisv/areafiftylan/web/banner/controller/BannerController.java

```diff
@@ -0,0 +1,112 @@
+    }
+
+    /**
+     * Get all banners in the database.
+     *
+     * @return all banners
+     */
+    @PreAuthorize("hasRole('ADMIN')")
+    @GetMapping
+    public ResponseEntity<?> getBanners() {
+        Collection<Banner> banners = bannerService.getAllBanners();
+        return new ResponseEntity<>(banners, HttpStatus.OK);
+    }
+
+    /**
+     * Returns a banner where the current date (date when the request was made)
+     * lies within the range of its start- and end-date.
+     *
+     * @return a banner
+     */
+    @GetMapping("/current")
+    public ResponseEntity<?> getCurrentBanner() {
+        Banner currentBanner = bannerService.getCurrentbanner();
+        return new ResponseEntity<>(currentBanner, HttpStatus.OK);
+    }
+
+    /**
+     * Adds a new banner to the database. The date-range should be valid (e.g. there shouldn't be
+     * another entity covering that range already).
+     *
+     * @param banner banner to be added
+     * @return a response-entity
+     */
+    @PreAuthorize("hasRole('ADMIN')")
+    @PostMapping
+    public ResponseEntity<?> addBanner(@RequestBody Banner banner) {
+        banner = bannerService.addBanner(banner);
+        return createResponseEntity(HttpStatus.OK,
+                "Successfully created a banner with ID=" + banner.getId(), banner);
+    }
+
+    /**
+     * Updates the banner with the given Id. The date-range should be valid (e.g. there shouldn't be
+     * another entity covering that range already).
+     *
+     * @param bannerId Id of the banner to be updated
+     * @param banner   banner to be updated
+     * @return the updated banner
+     */
+    @PreAuthorize("hasRole('ADMIN')")
+    @PostMapping("/{bannerId}")
+    public ResponseEntity<?> updateBanner(@PathVariable Long bannerId, @RequestBody Banner banner) {
+        Banner updated = bannerService.update(bannerId, banner);
+        return createResponseEntity(HttpStatus.OK,
+                "Successfully updated the banner with ID=" + bannerId, updated);
+    }
+
+    /**
+     * Deletes a banner with the given ID from the database and returns a response-entity.
+     *
+     * @param bannerId id of the banner to be removed
+     * @return a response-entity
+     */
+    @PreAuthorize("hasRole('ADMIN')")
+    @DeleteMapping("/{bannerId}")
+    public ResponseEntity<?> deleteBanner(@PathVariable Long bannerId) {
+        bannerService.removeBanner(bannerId);
+        return createResponseEntity(HttpStatus.OK,
+                "Successfully removed the banner with ID=" + bannerId);
+    }
+
+    /**
+     * Deletes all banners from the database.
+     *
+     * @return a response-entity
+     */
+    @PreAuthorize("hasRole('ADMIN')")
+    @DeleteMapping
+    public ResponseEntity<?> deleteAllBanners() {
+        bannerService.deleteBanners();
+        return createResponseEntity(HttpStatus.OK,
+                "Successfully deleted all banners");
+    }
+
+    @ExceptionHandler(BannerNotFoundException.class)
```

Comment:
What is the difference between handling here and in the centralized handling class?

Reply:
The centralized class is for exceptions that occur in more than one controller. In this case, it's unlikely we'll be seeing a BannerNotFoundException outside of the BannerController so this is the correct placement.

Reply:
TIL, thanks 👍